pandas/core/internals/array_manager.py

```diff
@@ -0,0 +1,896 @@
+                arr = arr._data  # type: ignore[union-attr]
+            if isinstance(arr, np.ndarray):
+                arr = np.atleast_2d(arr)
+            block = make_block(arr, placement=slice(0, 1, 1), ndim=2)
+            applied = getattr(block, f)(**kwargs)
+            while isinstance(applied, list):
+                # ObjectBlock gives double nested result?, some functions give no list
+                applied = applied[0]
+            arr = applied.values
+            if isinstance(arr, np.ndarray):
+                arr = arr[0, :]
+            result_arrays.append(arr)
+
+        return type(self)(result_arrays, self._axes)
+
+    # TODO quantile
+
+    def isna(self, func) -> "ArrayManager":
+        return self.apply("apply", func=func)
+
+    def where(self, other, cond, align: bool, errors: str, axis: int) -> "ArrayManager":
+        if align:
+            align_keys = ["other", "cond"]
+        else:
+            align_keys = ["cond"]
+            other = extract_array(other, extract_numpy=True)
+
+        return self.apply_with_block(
+            "where",
+            align_keys=align_keys,
+            other=other,
+            cond=cond,
+            errors=errors,
+            axis=axis,
+        )
+
+    # TODO what is this used for?
+    # def setitem(self, indexer, value) -> "ArrayManager":
+    #     return self.apply_with_block("setitem", indexer=indexer, value=value)
+
+    def putmask(self, mask, new, align: bool = True, axis: int = 0):
+
+        if align:
+            align_keys = ["new", "mask"]
+        else:
+            align_keys = ["mask"]
+            new = extract_array(new, extract_numpy=True)
+
+        return self.apply_with_block(
+            "putmask",
+            align_keys=align_keys,
+            mask=mask,
+            new=new,
+            axis=axis,
+        )
+
+    def diff(self, n: int, axis: int) -> "ArrayManager":
+        return self.apply_with_block("diff", n=n, axis=axis)
+
+    def interpolate(self, **kwargs) -> "ArrayManager":
+        return self.apply_with_block("interpolate", **kwargs)
+
+    def shift(self, periods: int, axis: int, fill_value) -> "ArrayManager":
+        if fill_value is lib.no_default:
+            fill_value = None
+
+        if axis == 0 and self.ndim == 2:
+            # TODO column-wise shift
+            raise NotImplementedError
+
+        return self.apply_with_block(
+            "shift", periods=periods, axis=axis, fill_value=fill_value
+        )
+
+    def fillna(self, value, limit, inplace: bool, downcast) -> "ArrayManager":
+        # TODO implement downcast
+        inplace = validate_bool_kwarg(inplace, "inplace")
+
+        def array_fillna(array, value, limit, inplace):
+
+            mask = isna(array)
+            if limit is not None:
+                limit = libalgos.validate_limit(None, limit=limit)
+                mask[mask.cumsum() > limit] = False
+
+            # if not self._can_hold_na:
```

Comment:
can you make sure to document-or-delete any commented-out code like this

Reply:
Will add a todo here that this case can still be handled